app/controllers/posts_controller.rb

```diff
@@ -8,12 +8,18 @@ def create
     @user = User.find(current_user.id)
     @post = @user.posts.new(post_params)
 
-    redirect_to @user if @post.save
+    if @post.save
+      redirect_to @user
+      flash[:notice] = 'You have successfully created a post'
+    else
+      redirect_back(fallback_location: root_path)
+      flash[:alert] = 'Your post was not created'
+    end
   end
 
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts.all
+    @posts = Post.includes(comments: [:author])
   end
```

Comment:
- [ ] Here to solve N + 1 problem, you should get the user with their posts. Kindly refactor it to: `@user.posts.includes(:comments)`